src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -156,6 +245,113 @@ public abstract partial class JsonNamingPolicy
+        public bool TryGetSingle(out float value) { throw null; }
+        [System.CLSCompliantAttribute(false)]
+        public bool TryGetUInt16(out ushort value) { throw null; }
+        [System.CLSCompliantAttribute(false)]
+        public bool TryGetUInt32(out uint value) { throw null; }
+        [System.CLSCompliantAttribute(false)]
+        public bool TryGetUInt64(out ulong value) { throw null; }
+    }
+    public partial class JsonObject : System.Text.Json.JsonNode, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode>>, System.Collections.IEnumerable
+    {
+        public JsonObject() { }
+        public JsonObject(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode>> jsonProperties) { }
+        public System.Text.Json.JsonNode this[string key] { get { throw null; } set { } }
+        public System.Collections.Generic.ICollection<string> PropertyNames { get { throw null; } }
+        public System.Collections.Generic.ICollection<System.Text.Json.JsonNode> Values { get { throw null; } }
+        public void Add(System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode> jsonProperty) { }
+        public void Add(string propertyName, bool propertyValue) { }
+        public void Add(string propertyName, byte propertyValue) { }
+        public void Add(string propertyName, System.Collections.Generic.IEnumerable<System.Text.Json.JsonNode> propertyValue) { }
+        public void Add(string propertyName, double propertyValue) { }
+        public void Add(string propertyName, short propertyValue) { }
+        public void Add(string propertyName, int propertyValue) { }
+        public void Add(string propertyName, long propertyValue) { }
+        [System.CLSCompliantAttribute(false)]
+        public void Add(string propertyName, sbyte propertyValue) { }
+        public void Add(string propertyName, float propertyValue) { }
+        public void Add(string propertyName, string propertyValue) { }
+        public void Add(string propertyName, System.Text.Json.JsonArray propertyValue) { }
+        public void Add(string propertyName, System.Text.Json.JsonNode propertyValue) { }
+        [System.CLSCompliantAttribute(false)]
+        public void Add(string propertyName, ushort propertyValue) { }
+        [System.CLSCompliantAttribute(false)]
+        public void Add(string propertyName, uint propertyValue) { }
+        [System.CLSCompliantAttribute(false)]
+        public void Add(string propertyName, ulong propertyValue) { }
+        public void AddRange(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode>> jsonProperties) { }
+        public bool ContainsProperty(string propertyName) { throw null; }
+        public System.Collections.Generic.IEnumerable<System.Text.Json.JsonNode> GetAllPrimaryTypedValues() { throw null; }
```

Comment:
What do you mean by PrimaryTyped?

Reply:
So I thought about the following alternatives: 
GetAllValues, GetAllPrimaryTypes, GetAllPrimaryValues, GetAllLeafValues, GetAllInnermostValues, GetAllLeaves.

I didn't choose just "Values", because `JsonObject` and `JsonArray` are also called values in Json grammar. I didn't choose "Leaves", because it is implementation-specific (however developers might be used to thinking of Json as a tree). I decided to use "PrimaryTyped", because it more or less explains what kind of values it will return (JsonString, JsonNumber, JsonBool, JsonNull, not: JsonArray or JsonObject), but I can change it to a another proposition or something completely different (or remove this method at all).

Reply:
I guess my question is why would someone want to get the property values excluding array and objects? Correct me if I'm wrong but I believe that in most cases people will most likely want to get all property values, not only the ones that are not collections.

Reply:
Yeah, I don't know that it makes any sense to filter on the primitive types or complex types.  I'd like to understand an actual scenario for it.

Reply:
The scenario I was thinking about is `TestAquiringAllPrimaryTypedValues`.

Reply:
But I wasn't sure if we need so specialized methods.